src/bootstrap/lib.rs

```diff
@@ -766,6 +775,12 @@ impl Build {
         if &*target == "i686-pc-windows-gnu" {
             base.push("-fno-omit-frame-pointer".into());
         }
+
+        if self.cc(target).ends_with("clang") {
```

Comment:
Why only Clang? Gcc should accept the same flag.

Reply:
Oh sure! I was unaware of that

(eventually we'll need to figure out how to pass this to clang-cl as well)

Reply:
clang-cl seems to accept it if you stick `-Xclang` before it, from cursory testing:
```
c:\build>c:\Users\ted\.mozbuild\clang\bin\clang-cl.exe -Fohello.obj -c c:\build\hello.c -Ic:\build -Z7 -Xclang -fdebug-prefix-map=c:\build=\src

$ stringsext hello.obj | rg src
\src\src\hello.c
```